src/systemd/src/basic/path-util.c

```diff
@@ -372,9 +375,14 @@ char *path_simplify(char *path, bool kill_dots) {
                 *(t++) = *f;
         }
 
-        /* Special rule, if we are talking of the root directory, a trailing slash is good */
-        if (absolute && t == path)
-                *(t++) = '/';
+        /* Special rule, if we stripped everything, we either need a "/" (for the root directory)
+         * or "." for the current directory */
+        if (t == path) {
+                if (absolute)
+                        *(t++) = '/';
+                else
+                        *(t++) = '.';
+        }
```

Comment:
Won't this create a conflict when the code changes in systemd and we synchronize it? We should probably have the fix accepted in systemd before.

Reply:
it's already fixed in systemd upstream. I added a URL to the systemd commit to the commit message.